### PR TITLE
[Simplify] Remove custom source set implementation in favour of using…

### PR DESCRIPTION
**Pull request by Nava2**
… extension directly attached to `sourceSet`

This removes the custom `AssertJSourceSet` which was not adding value with the new functionality in modern gradle that generates methods to configure extensions for those that they are attached to. For example, the `assertJ {}` closure is handled by the `extension` rather than adding a new `SourceSet` type.

This was extracted from #31

### CODE REVIEW

File: src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy [hunk 1]
```diff
@@ -71,55 +68,42 @@ class AssertJGeneratorGradlePlugin implements Plugin<Project> {
...
 
         // Create a new task for the source set
-        AssertJGenerationTask generationTask = project.tasks.findByName(generateTaskName) as AssertJGenerationTask
-
-        if (!generationTask) {
-            generationTask = project.tasks.create(generateTaskName, AssertJGenerationTask) {
-                description = "Generates AssertJ assertions for the ${sourceSet} sources."
-                generationClasspath = sourceSet.runtimeClasspath // Get the classes used when creating the ClassLoader for
-                                                                 // Generation
+        def generationTask = project.tasks.register(
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy]: The `register` method does the same thing that I had custom-written back in the day.

File: src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy [hunk 2]
```diff
@@ -71,55 +68,42 @@ class AssertJGeneratorGradlePlugin implements Plugin<Project> {
...
 
         logger.info("generationTask: ${generateTaskName}, sourceSet: ${sourceSet}")
 
         // Create a new task for the source set
-        AssertJGenerationTask generationTask = project.tasks.findByName(generateTaskName) as AssertJGenerationTask
-
-        if (!generationTask) {
-            generationTask = project.tasks.create(generateTaskName, AssertJGenerationTask) {
-                description = "Generates AssertJ assertions for the ${sourceSet} sources."
-                generationClasspath = sourceSet.runtimeClasspath // Get the classes used when creating the ClassLoader for
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy]: All of this `init` code moved into the constructor for the `AssertJGenerationTask`.

File: src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy [hunk 3]
```diff
@@ -135,10 +119,8 @@ class AssertJGeneratorGradlePlugin implements Plugin<Project> {
                 }
 
                 // With the test task, we add it to the _test_ source set
-                testSourceSet.allSource.source(assertJSS.assertJ)
-
-                testSourceSet.java.srcDirs += generationTask.outputDir
-                project.tasks.findByName(testSourceSet.compileJavaTaskName).dependsOn generationTask
+                testSourceSet.java.srcDir(generationTask.map { it.outputDir })
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/AssertJGeneratorGradlePlugin.groovy]:
```suggestion
                testSourceSet.java.srcDir(generationTask.flatMap { it.outputDir })
```

File: src/main/groovy/org/assertj/generator/gradle/internal/tasks/config/DefaultAssertJGeneratorOptions.groovy [hunk 4]
```diff
@@ -37,43 +37,24 @@ class DefaultAssertJGeneratorOptions implements AssertJGeneratorOptions, Seriali
     private Templates templates
     protected EntryPointGeneratorOptions _entryPoints
 
-    protected String outputDir
+    final DirectoryProperty outputDir
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/internal/tasks/config/DefaultAssertJGeneratorOptions.groovy:41]: This change was required to remove the custom source set and properly handle when it changes.

File: src/main/groovy/org/assertj/generator/gradle/internal/tasks/config/DefaultAssertJGeneratorOptions.groovy [hunk 5]
```diff
@@ -117,14 +98,14 @@ class DefaultAssertJGeneratorOptions implements AssertJGeneratorOptions, Seriali
...
-        s.writeObject(outputDir)
+        s.writeObject(outputDir.getAsFile().get())
         s.writeBoolean(skip)
         s.writeObject(_entryPoints)
         s.writeObject(templates)
     }
 
     private void readObject(ObjectInputStream s) throws IOException, ClassNotFoundException {
-        this.outputDir = (String) s.readObject()
+        this.outputDir.set(s.readObject() as File)
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/internal/tasks/config/DefaultAssertJGeneratorOptions.groovy]: I'm like 95% sure this doesn't work and isn't needed. Will remove.

File: src/main/groovy/org/assertj/generator/gradle/tasks/AssertJGenerationTask.groovy [hunk 6]
```diff
@@ -21,47 +21,60 @@ import org.assertj.assertions.generator.description.converter.ClassToClassDescri
...
 
     @InputFiles
-    List<File> getTemplateFiles() {
-        assertJOptions.templates.files
-    }
+    @Classpath
+    final ListProperty<File> templateFiles
+
+    @Input
+    final ListProperty<String> templateStrings
```
Comment (Nava2) [inline src/main/groovy/org/assertj/generator/gradle/tasks/AssertJGenerationTask.groovy]: oops this change shouldnt' be here

**Review by scordio — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -144,7 +144,7 @@ sourceSets {
     // turn on assertJ generation
     assertJ { 
       // default: generated-srcs/${SOURCE_SET_NAME_TAG}/java
-      outputDir = "src-gen/${SOURCE_SET_NAME_TAG}/java"
+      outputDir = file("src-gen/${SOURCE_SET_NAME_TAG}/java")
```
Comment (Nava2) [inline README.md:147]:
```suggestion
      outputDir = file("src-gen/main-test/java")
```

**Pull request merged**